src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,163 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    public MyLinkedList() {
+
+    }
+
     @Override
     public void add(T value) {
+        Node<T> newNode;
+        if (size == 0) {
+            newNode = new Node<>(null, value, null);
+            head = newNode;
+        } else {
+            newNode = new Node<>(tail, value, null);
+            tail.next = newNode;
+        }
+        tail = newNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        Node<T> nodeToAdd;
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("index: " + index + " Out of bound size : " + size);
+        }
+        if (index == size) {
+            add(value);
+            return;
+        }
+        Node<T> wantedNode = findNodeByIndex(index);
+        if (index == 0) {
+            nodeToAdd = new Node<>(null, value, wantedNode);
+            head = nodeToAdd;
+        } else {
+            nodeToAdd = new Node<>(wantedNode.prev, value, wantedNode);
+            wantedNode.prev.next = nodeToAdd;
+        }
+        wantedNode.prev = nodeToAdd;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T listElement : list) {
+            add(listElement);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        indexCheck(index);
+        Node<T> wantedNode = findNodeByIndex(index);
+        return wantedNode.value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        indexCheck(index);
```

Comment:
It is better to move this method to findNodeByIndex, and only call it here. Do this in other places where you call findNodeByIndex